workflow/controller/controller.go

```diff
@@ -450,7 +450,8 @@ func (wfc *WorkflowController) processNextPodCleanupItem(ctx context.Context) bo
 	}()
 	if err != nil {
 		logCtx.WithError(err).Warn("failed to clean-up pod")
-		if errorsutil.IsTransientErr(err) {
+		podGCMaxRequeues := env.LookupEnvIntOr("POD_GC_TRANSIENT_ERROR_MAX_REQUEUES", 2)
```

Comment:
Are all of these env flags getting documented somewhere? If not, they should be

Reply:
I don't think so. Yes I agree that we should document all the variables. I can help summarizing them when I find some time later but perhaps in a separate PR.